### PR TITLE
Prep for v0.14.0 - reset and modernize repo and tooling

### PR DESCRIPTION
**Pull request by Jonic**
This is the start of a 0.14.0 release. This PR does not fix any of the deprecation warnings that we've been seeing, it just resets and modernises the tooling in the repo. When this PR is merged, I can create another branch that specifically fixes only the Ruby 2.7 deprecations, so they can be handled in isolation.

This PR contains the following changes:

* Update the Gemspec to reflect the fact that it's our repo/Gem
* Update the Gem's dev dependencies
* Update the Ruby version to match Panacea
* Reset the Rubocop config with up-to-date rules
* Applies Rubocop autocorrections to lib and spec code
* Updates RSpec config
* Adds Simplecov
* Adds Dependabot config
* Adds RSpec and Linters CI actions
* Removes Travis CI and Codeclimate config
* Adds Guard and config
* Adds generated binstubs

For the purposes of clarity - anything changed in `lib` or `spec` was more than likely generated by the updated Rubocop rules. You can probably ignore the changes in those directories. Any changes to the specs have changed the formatting only, and not the actual code under test.

Remarkably, the code coverage of the `lib` directory is 98.16%, so given that the test suite still passes with no new or modified specs, I'm confident that the changes Rubocop has made to the code have not changed any functionality.

These changes are mostly about making this repo easier to work with. When this has merged we can focus on the deprecation warning fixes, and THEN tag and release the 0.14.0 version for installation in Panacea.

### CODE REVIEW

**Comment by ronanduddy** [comment]
I believe this is the Jira ticket: https://healiosuk.atlassian.net/browse/BC-29

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +0,0 @@
-language: ruby
-rvm:
- - "2.2.10"
- - "2.3.7"
- - "2.4.4"
- - "2.5.1"
-script: bundle exec rspec
```
Comment (ronanduddy) [inline .travis.yml:7]: Hmmm, what CI do we have for this forked gem now?

File: .github/workflows/rspec.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Ruby and install gems
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+
+      - name: Run tests
+        run: bin/rspec
```
Comment (ronanduddy) [inline .github/workflows/rspec.yml:26]: Excellent 👍, we're using GitHub actions to replace travis

File: Gemfile.lock [hunk 3]
```diff
@@ -1,138 +1,216 @@
...
-    rectify (0.13.0)
-      activemodel (>= 4.1.0)
-      activerecord (>= 4.1.0)
-      activesupport (>= 4.1.0)
-      virtus (~> 1.0.5)
-      wisper (>= 1.6.1)
+    rectify (0.14.0)
+      activemodel (~> 7.0)
+      activerecord (~> 7.0)
+      activesupport (~> 7.0)
```
Comment (ronanduddy) [inline Gemfile.lock:7]: I wonder does the rails-gem upgrade cause any issues?

File: .tool-versions [hunk 4]
```diff
@@ -26,16 +26,13 @@
 #
 # 4. Add plugins
 #    `asdf plugin add bundler`
-#    `asdf plugin add nodejs`
 #    `asdf plugin add ruby`
-#    `asdf plugin add yarn`
```
Comment (ronanduddy) [inline .tool-versions]: 👍

**Review by svpersteve — APPROVED** [review]

File: .rubocop.yml [hunk 5]
```diff
@@ -1,36 +1,21 @@
...
+    - "spec/db/schema.rb"
+  NewCops: "enable"
+  SuggestExtensions: false
+  TargetRubyVersion: 2.7.8
 
 Style/Documentation:
   Enabled: false
-
-Style/HashSyntax:
-  EnforcedStyle: hash_rockets
```
Comment (ronanduddy) [inline .rubocop.yml:12]: 👍 for addressing

File: .rubocop.yml [hunk 6]
```diff
@@ -1,36 +1,21 @@
...
-
-Metrics/ClassLength:
-  Max: 115
-  Exclude:
-    - 'spec/**/*_spec.rb'
-
-Metrics/BlockLength:
-  Exclude:
-    - 'spec/**/*_spec.rb'
+  EnforcedStyle: "double_quotes"
```
Comment (ronanduddy) [inline .rubocop.yml:22]: 👍 keeping

File: .rubocop_todo.yml [hunk 7]
```diff
@@ -5,64 +5,3 @@
...
-Style/NumericPredicate:
-  Exclude:
-    - 'spec/**/*'
-    - 'lib/rectify/form.rb'
-
-# Offense count: 1
-# Configuration parameters: AllowHeredoc, AllowURI, URISchemes, IgnoreCopDirectives, IgnoredPatterns.
-# URISchemes: http, https
-Metrics/LineLength:
-  Max: 100
```
Comment (ronanduddy) [inline .rubocop_todo.yml:68]: 👍 for addressing

File: rectify.gemspec [hunk 8]
```diff
@@ -1,41 +1,29 @@
...
+  spec.metadata["github_repo"] = "ssh://github.com/healiosuk/rectify"
+  spec.metadata["homepage_uri"] = spec.homepage
+  spec.metadata["rubygems_mfa_required"] = "true"
+  spec.metadata["source_code_uri"] = spec.homepage
+
+  spec.add_dependency "activemodel", "~> 7.0"
+  spec.add_dependency "activerecord", "~> 7.0"
+  spec.add_dependency "activesupport", "~> 7.0"
+  spec.add_dependency "virtus", "~> 1.0"
+  spec.add_dependency "wisper", "~> 1.6"
```
Comment (ronanduddy) [inline rectify.gemspec:28]: Rejigging the gemspec-gemfile here

File: Rakefile [hunk 9]
```diff
@@ -6,19 +6,10 @@ require "active_record"
...
-
   desc "Migrate the database"
   task :migrate do
     ActiveRecord::Base.establish_connection(db_config)
-
-    if activerecord_below_5_2?
-      ActiveRecord::Migrator.migrate("spec/db/migrate")
-    else
-      ActiveRecord::MigrationContext.new("spec/db/migrate").migrate
-    end
```
Comment (ronanduddy) [inline Rakefile:21]: We are removing this so as to not support <5.2?

File: spec/db/schema.rb [hunk 10]
```diff
@@ -2,42 +2,41 @@
...
+# db:schema:load`. When creating a new database, `bin/rails db:schema:load` tends to
+# be faster and is potentially less error prone than running all of your
+# migrations from scratch. Old migrations may fail to apply correctly if those
+# migrations use external dependencies or application code.
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2018_05_31_090029) do
-
+ActiveRecord::Schema[7.0].define(version: 2018_05_31_090029) do
```
Comment (ronanduddy) [inline spec/db/schema.rb:13]: Unsure if we should update the version as well?

File: spec/db/migrate/20160407175608_add_user.rb [hunk 11]
```diff
@@ -1,10 +1,10 @@
 class AddUser < ActiveRecord::Migration[4.2]
   def change
     create_table :users do |t|
-      t.string :first_name, :null => false, :default => ""
-      t.integer :age,       :null => false, :default => 0
+      t.string :first_name, null: false, default: ""
+      t.integer :age,       null: false, default: 0
```
Comment (ronanduddy) [inline spec/db/migrate/20160407175608_add_user.rb:5]: This commit moves away from using the hash rocket syntax

**Review by ronanduddy — APPROVED** [review]
LGTM - just some observations for other folks

**Comment by ronanduddy** [comment]
I was thinking if it made sense to release this as `0.13.1` rather than `0.14.0` however, I see the rails upgrade as being a major improvement.

File: Rakefile (hunk 9 above)
Comment (Jonic) [reply Rakefile:21]: We're only ever going to run this on Panacea, which is Rails 7, so I figured it wouldn't hurt to take that out.

File: spec/db/schema.rb (hunk 10 above)
Comment (Jonic) [reply spec/db/schema.rb:13]: This was generated by the `db:migrate` command. The updated Gems produce the updated ActiveRecord::Schema version number.

File: Rakefile (hunk 9 above)
Comment (ronanduddy) [reply Rakefile:21]: I agree 👍

**Pull request merged**